### PR TITLE
feat: add dockerfile schematics and jest support

### PR DESCRIPTION
**Pull request by w3dani**
## What’s the focus of this PR
Two main changes are implemented in this branch:
- Add a new schematics for Dockerfile. It requires to select the type of build (basic node or nestjs) and node version and it generates the dockerfile and the .dockerignore
- Replaces `jasmine` with `jest` and fixes some tests that were not passing

## How to review this PR
Dockerfile schematics:
- `npm run build -w ./packages/@guidesmiths/cuckoojs-schematics` 
- then `schematics @guidesmiths/cuckoojs-schematics:dockerfile `

Tests:
- `npm run test -w ./packages/@guidesmiths/cuckoojs-schematics`

Before submitting this PR, I made sure:
- [x] The code builds clean without any errors or warnings
- [x] I'm using our guidelines
- [x] I've added unit tests if feasible

### CODE REVIEW

File: packages/@guidesmiths/cuckoojs-schematics/src/gitignore/gitignore.factory.spec.ts [hunk 1]
```diff
@@ -8,9 +8,11 @@ describe('gitignore', () => {
...
     const runner = new SchematicTestRunner('schematics', collectionPath);
     const tree = await runner
-      .runSchematicAsync('gitignore', {}, Tree.empty())
+      .runSchematicAsync('gitignore', { directory: '.'}, Tree.empty())
       .toPromise();
 
-    expect(tree.files).toEqual([]);
+    expect(tree.files).toEqual([
+      '/.gitignore'
+    ]);
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-schematics/src/gitignore/gitignore.factory.spec.ts:16]: Thanks!!! We didn't ever start with tests ❤️

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/dockerfile.factory.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Tree } from '@angular-devkit/schematics';
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/dockerfile.factory.spec.ts:1]: 👏🏼

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          { "value": 18,  "label": "18 (recommended)" },
+          { "value": 16,  "label": "16" },
+          { "value": 14,  "label": "14" },
+          { "value": 12,  "label": "12 (not recommended)" }
+        ],
+        "default": 18
+      }
+    }
+  },
+  "required": ["directory"]
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json:47]: Should `buildType` be also required? If not, there should be a default value for it, shouldn't it?

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ENV NODE_ENV production
+
+USER node
+WORKDIR /home/node
+
+COPY --from=builder --chown=node:node /home/node/package*.json ./
+COPY --from=builder --chown=node:node /home/node/node_modules/ ./node_modules/
+COPY --from=builder --chown=node:node /home/node/dist/ ./dist/
+
+CMD ["node", "dist/server.js"]
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile:28]: Do we need to be able to configure the command to be run or will it be always the same for nodejs or nestjs apps?

**Review by inigomarquinez — COMMENTED** [review]
Awesome work @w3dani 

I've added a couple of questions

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile (hunk 4 above)
Comment (w3dani) [reply packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile:28]: I'd keep it this way and we we start using it, we could iterate over this

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json (hunk 3 above)
Comment (w3dani) [reply packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json:47]: true, and I'd add nodeVersion as required as well

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile (hunk 4 above)
Comment (inigomarquinez) [reply packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/files/nestjs/Dockerfile:28]: Sounds good!

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json [hunk 5]
```diff
@@ -44,5 +44,5 @@
       }
     }
   },
-  "required": ["directory"]
+  "required": ["directory", "buildType", "nodeVersion"]
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json:47]: @w3dani , perhaps you don't need to include "nodeVersion" as it has a default value. What do you prefer?

**Review by inigomarquinez — COMMENTED** [review]
One last comment

File: packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json (hunk 5 above)
Comment (w3dani) [reply packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json:47]: I think that's more a philosophical discussion. Is "having a default value" exclusive from "being required"? I think having both values is more meaningful, but it's ok for me to remove it from required list.

Comment (inigomarquinez) [reply packages/@guidesmiths/cuckoojs-schematics/src/dockerfile/schema.json:47]: Understood. Let's keep it as it is for now.

**Review by inigomarquinez — APPROVED** [review]
LGTM!

**Comment by inigomarquinez** [comment]
@w3dani , can you solve the conflicts that have appeared after merging your previous PR? Thanks!

**Review by inigomarquinez — APPROVED** [review]
LGTM!

**Review by neodmy — APPROVED** [review]

**Review by ardguezsoc — APPROVED** [review]

**Review by dustytrinkets — APPROVED** [review]

**Pull request merged**